src/jobs/deploy-lambda.yml

```diff
@@ -1,9 +1,59 @@
 description: >
   This job deploys an application to aws lambda.
 
+parameters:
+  aws-region:
+    type: env_var_name
+    default: AWS_REGION_N_VA
+    description: |
+      Specify the region for aws
+
+  aws-access-key-id:
+    type: env_var_name
+    default: MACHINE_AWS_ECS_ACCESSKEY
+    description: |
+      Access key for MACHINE_AWS_ECS
+
+  aws-secret-access-key-id:
+    type: env_var_name
+    default: MACHINE_AWS_ECS_SECRETKEY
+    description: |
+      Secret Key for Machine_AWS_ECS
+
+  tag:
+    type: string
+    description: |
+      The semantic version of the project to be deployed
+
+  slack-mentions:
+    type: env_var_name
+    default: MENTIONS
+    description: |
+      MENTIONS stores the environment variable name for the Slack context.
+
+  lambda-function-name:
+    type: string
+    description: |
+      The name of the lambda function to be updated.
+
 executor: node/default
 
 steps:
+  - checkout
+  - aws-cli/setup:
+      aws-access-key-id: <<parameters.aws-access-key-id>>
+      aws-region: <<parameters.aws-region>>
+      aws-secret-access-key: <<parameters.aws-secret-access-key-id>>
+  - tag-repo:
+      tag: <<parameters.tag>>
   - run:
-      name: "Not Yet Implemented"
-      command: echo "Not Yet Implemented"
+      name: Update Lambda Function code
+      command: >
+        aws lambda update-function-code --function-name <<parameters.lambda-function-name>>
```

Comment:
There are a couple of ways to update a lambda function. Uploading a zip or a docker image. You should talk to chris and see if we want to implement both or just one. Here is the aws cli reference: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/update-function-code.html